crates/rpc/rpc/src/trace.rs

```diff
@@ -257,15 +257,16 @@ where
         }
 
         // fetch all blocks in that range
-        let blocks = self.provider().block_range(start..=end)?;
+        let blocks = self.provider().block_range_with_senders(start..=end)?;
 
         // find relevant blocks to trace
         let mut target_blocks = Vec::new();
-        for block in blocks {
+        for block_with_senders in blocks {
             let mut transaction_indices = HashSet::new();
             let mut highest_matching_index = 0;
-            for (tx_idx, tx) in block.body.iter().enumerate() {
-                let from = tx.recover_signer().ok_or(BlockError::InvalidSignature)?;
+
+            for (tx_idx, tx) in block_with_senders.block.body.iter().enumerate() {
+                let from = block_with_senders.senders[tx_idx];
```

Comment:
this should make use of the iterator functions in BlockWithSenders

Reply:
Done!